### DISCUSSION TITLE
Audio output silenced after G.C.

### DISCUSSION THREAD

**Discussion by Psi58** [GrandOrgue/grandorgue#1863] [Help] [answered]
I've converted a HW organ using ODFEdit and doing some tidying of delimiters (/ and \) and file paths. Whilst the resulting ODF initially works in GO it subsequently starts behaving strangely.

**From a working state, I press the G.C. button in the Combination Setter panel, after which I cannot get any sound at all, whatever stops I select.**

What can be causing this? It looks very much as if that G.C. function is changing more than the selected stops. But I've spent a day trying to work out what's going on, without success.

I suspect that the problem is in GO, not ODFEdit, as the converted file works initially. But I can't be certain, and maybe ODFEdit is generating something incorrect that nevertheless parses OK in GO.

Any suggestions are welcome. Thanks.

**Comment by eturpault** [comment, marked as answer]
I have found the reason of this bad behavior. In the sample set "Pipeloops Romantic Village Church Organ" the stops are controlled by a strange network of switches, some of them are engaged by default and they cannot be controlled from a panel (and probably they are not affected by the GC in HW as they are not clickable). When the G.C. button of GO is pressed, these non controllable switches become disengaged (OdfEdit has given them the attribute GCstate=0), so the stops (which the activation state is a logical AND between a controllable switch and a non controllable switch) cannot be engaged anymore.
For the next version of OdfEdit I have made a modification to ignore such switches which cannot be controlled by the user.
Here is an ODF generated with this modification : [Pipeloops Romantic Village Church Organ.zip](https://github.com/GrandOrgue/grandorgue/files/14890438/Pipeloops.Romantic.Village.Church.Organ.zip)

**Reply by Psi58** [reply]
Thank you! That explanation makes perfect sense. 
I wonder why the designer includes switches that cannot be controlled at a user level? Perhaps the HW ODF was created using another organ as a template, and not cleaned-up afterwards...?
I'm very grateful for your quick response, the modified ODF works fine.

**Comment by ClarkeBrunt** [comment]
I wrote about a similar problem with the well-known Friesach organ here #1407 - in that case it's just various action-noises which are permanently silenced by G.C. rather than the whole organ. In that case, GCstate was completely absent from the 'stops' in question, which according to the docs means 'implementation defined' (seems to mean 'turn off' in practice). And you can probably find (and turn back on) the offending 'stops' in the 'MIDI Objects' list (menu bar Audio/Midi), at least while experimenting.

**Reply by Psi58** [reply]
Interesting - thank you. 
Perhaps GO's ODF check should also verify that any object that GC can modify is surfaced at least once for the user to control?
I hadn't realised that the buried stops could be toggled in 'MIDI Objects' - thanks for that tip. Mind you, as @eturpault noted, this particular organ has a complex network of switches, quite difficult to visualise clearly.

**Comment by eturpault** [comment]
Yes very interesting, thanks @ClarkeBrunt for having reminded this old discussion. Now I understand why sometimes I had no audible keys noises with HW sample sets converted to GO by OdfEdit, and I was unable to understand the reason. 
I reproduced this issue with Friesach and Giubiasco of Piotr Grabowski as well.

In next release of OdfEdit, for the HW to GO conversion, the Stop sections generating keys noises will have GCState=-1, it will permit to never loose the keys noises as said in your discussion. I tested it and it works !

@oleg68 : wouldn't it be more predictable if GCState had a default value at -1 instead of "implementation defined" which creates unexpected behaviors ? Or in the help to detail what "implementation defined" means ? 
I see that the warning "Unused ODF entry" is generated by GO for the Stop sections having GCState=-1 when they are controlled by a Switch.

**Reply by eturpault** [reply]
Looking at the source code of GrandOrgue in GODrawstop::Load, if I read well GCState is systematically set by default at 0, so the help should indicate "default : 0" instead of "default: implementation defined", isn't it ?
